src/ui/SmartSnippet/SmartSnippetSuggestions.ts

```diff
@@ -0,0 +1,88 @@
+import { QueryEvents, IQuerySuccessEventArgs } from '../../events/QueryEvents';
+import { IPartialQuestionAnswerResponse, IQuestionAnswerResponse } from '../../rest/QuestionAnswerResponse';
+import { $$, Dom } from '../../utils/Dom';
+import { uniqueId, isEqual } from 'underscore';
+import { SmartSnippetCollapsibleSuggestion } from './SmartSnippetCollapsibleSuggestion';
+import { l } from '../../strings/Strings';
+
+const BASE_CLASSNAME = 'coveo-smart-snippet-suggestions';
+const QUESTIONS_LIST_CLASSNAME = `${BASE_CLASSNAME}-questions`;
+const QUESTIONS_LIST_TITLE_CLASSNAME = `${QUESTIONS_LIST_CLASSNAME}-title`;
+
+export interface ISmartSnippetSuggestionsOptions {}
+
+export class SmartSnippetSuggestions extends Component {
+  static ID = 'SmartSnippetSuggestions';
+
+  static options: ISmartSnippetSuggestionsOptions = {};
+
+  static doExport = () => {
+    exportGlobally({
+      SmartSnippetSuggestions
+    });
+  };
+
+  private readonly titleId = uniqueId(QUESTIONS_LIST_TITLE_CLASSNAME);
+  private contentLoaded: Promise<any>;
+  private title: Dom;
+  private questionAnswers: Dom;
+  private renderedQuestionAnswer: IQuestionAnswerResponse;
+
+  constructor(public element: HTMLElement, public options?: ISmartSnippetSuggestionsOptions, bindings?: IComponentBindings) {
+    super(element, SmartSnippetSuggestions.ID, bindings);
+    this.bind.onRootElement(QueryEvents.deferredQuerySuccess, (data: IQuerySuccessEventArgs) => this.handleQuerySuccess(data));
+  }
+
+  public get loading() {
+    return this.contentLoaded;
+  }
+
+  private handleQuerySuccess(data: IQuerySuccessEventArgs) {
+    const questionAnswer = data.results.questionAnswer;
+    if (questionAnswer && questionAnswer.relatedQuestions.length) {
+      if (this.renderedQuestionAnswer && isEqual(questionAnswer, this.renderedQuestionAnswer)) {
+        return;
+      }
+      this.detachContent();
+      this.element.appendChild((this.title = this.buildTitle()).el);
+      this.element.appendChild((this.questionAnswers = this.buildQuestionAnswers(questionAnswer.relatedQuestions)).el);
+    } else {
+      this.detachContent();
+    }
+    this.renderedQuestionAnswer = questionAnswer;
+  }
+
+  private detachContent() {
+    this.title && this.title.detach();
+    this.questionAnswers && this.questionAnswers.detach();
+    this.title = this.questionAnswers = null;
+  }
+
+  private buildTitle() {
+    return $$('span', { className: QUESTIONS_LIST_TITLE_CLASSNAME, id: this.titleId }, l('SuggestedQuestions'));
+  }
+
+  private buildQuestionAnswers(questionAnswers: IPartialQuestionAnswerResponse[]) {
+    const innerCSS = this.getInnerCSS();
+    const answers = questionAnswers.map(questionAnswer => new SmartSnippetCollapsibleSuggestion(questionAnswer, innerCSS));
+    const container = $$(
+      'ul',
+      { className: QUESTIONS_LIST_CLASSNAME, ariaLabelledby: this.titleId },
+      ...answers.map(answer => answer.build())
```

Comment:
Similar comment here on making sure rendering the answers as html cannot be used for XSS attacks. Never trust your neighbors :)

Reply:
Never let the clients trust themselves!

Reply:
This line seems safe. What could be vulnerable to an XSS attack is line 91-98 in `SmartSnippetCollapsibleSuggestion`:
```ts
  private buildAnswerSnippetContent(innerHTML: string, style?: HTMLStyleElement) {
    const snippet = $$('div', {}, innerHTML);
    const container = $$('div', {}, snippet);
    if (style) {
      container.append(style);
    }
    return container;
  }
```

However, the question and answer text are returned by the API, and if we have to worry about XSS here then all our clients who call the API directly need to worry about XSS too. Currently, Question Answering filters out all HTML attributes and tags that aren't part of a very short allow-list.